### PR TITLE
Allow changing the "color scheme" in the preview pane.

### PR DESCRIPTION
**Pull request by pykaso**

### CODE REVIEW

File: UIViewPreviewProvider.swift [hunk 1]
```diff
@@ -48,6 +48,10 @@ public protocol UIViewPreviewProvider {
     
     /// Array of `Preview` to be displayed in the Xcode canvas  pane `(⌥ + ⌘ + return)`
     static var uiPreviews: [Preview] { get }
+    
+    /// `ColorScheme` to be used in the Xcode canvas pane
+    @available(iOS 13.0, *)
+    static var colorScheme: ColorScheme { get }
```
Comment (mrabiciu) [inline UIViewPreviewProvider.swift]: This means we'd need to create two different `UIViewPreviewProvider` in order to preview both light mode and dark mode.

Lets make this an array instead:
```swift
static var colorSchemes: [ColorScheme] { get }
```

which we can default to:
```swift
public static var colorSchemes: [ColorScheme] {
    [.light, .dark]
}
```

File: UIViewPreviewProvider.swift [hunk 2]
```diff
@@ -100,6 +104,7 @@ extension UIViewPreviewProvider {
         
         return ForEach(identifiablePreviews) { preview in
             swiftUIView(from: preview.item)
+                .environment(\.colorScheme, colorScheme)
         }
     }
```
Comment (mrabiciu) [inline UIViewPreviewProvider.swift]: Then down here we can iterate over all the color schemes and make a preview for each of them
```swift
    public static var previews: some View {
        
        let identifiablePreviews = uiPreviews.map {
            IdentifiableBox($0)
        }
        
        let identifieableSchemes = colorSchemes.map {
            IdentifiableBox($0)
        }
        
        return ForEach(identifiablePreviews) { preview in
            ForEach(identifieableSchemes) { scheme in
                swiftUIView(from: preview.item)
                    .environment(\.colorScheme, scheme.item)
            }
        }
    }
```

**Review by mrabiciu — CHANGES_REQUESTED** [review]
I think this makes sense, but we can generalize it a bit more to allow showing light and dark mode side by side automatically.

What do you think?

**Comment by pykaso** [comment]
> I think this makes sense, but we can generalize it a bit more to allow showing light and dark mode side by side automatically.
> 
> What do you think?

I like this idea. Nice job.

**Review by mrabiciu — APPROVED** [review]

**Comment by mrabiciu** [comment]
Thanks for contributing!

**Pull request merged**